### PR TITLE
Bring back live host change

### PR DESCRIPTION
**Pull request by agocke**
The previous problem seems to be the usage of RuntimeFiles as the ItemGroup for the host files instead of NativeRuntimeAsset, and the lack of `PackOnly`, which prevents hostfxr from being included in the shared framework package.

### CODE REVIEW

File: src/libraries/externals.csproj [hunk 1]
```diff
@@ -21,23 +18,23 @@
...
   <ItemGroup Condition="'$(DotNetBuildFromSource)' != 'true'">
     <PackageReference Include="Microsoft.DiaSymReader.Native"
                       Version="$(MicrosoftDiaSymReaderNativeVersion)" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetsMobile)' != 'true' and '$(UseLiveBuiltDotNetHost)' != 'true'">
-    <PackageReference Include="Microsoft.NETCore.DotNetHost"
-                      Version="$(MicrosoftNETCoreDotNetHostVersion)" />
-    <PackageReference Include="Microsoft.NETCore.DotNetHostPolicy"
-                      Version="$(MicrosoftNETCoreDotNetHostPolicyVersion)" />
```
Comment (am11) [inline src/libraries/externals.csproj:34]: These are not used anywhere else, we can delete them from version files:
https://github.com/dotnet/runtime/blob/17a4eca42337109c4f916a023d97df12b50b5d59/eng/Versions.props#L77-L78
https://github.com/dotnet/runtime/blob/17a4eca42337109c4f916a023d97df12b50b5d59/eng/Version.Details.xml#L213-L220

**Comment by agocke** [comment]
also @jkoritzinsky

**Comment by ViktorHofer** [comment]
@agocke in general looks good but can you please point me to the fix (in this change) for the previous regression?

File: eng/liveBuilds.targets [hunk 2]
```diff
@@ -118,6 +118,16 @@
       <MonoIncludeFiles Include="$(MonoArtifactsPath)\include\**\*.*" />
     </ItemGroup>
 
+    <!-- Host files. Mobile uses a different hosting model, so we don't include the .NET host components there. -->
+    <ItemGroup Condition="'$(TargetsMobile)' != 'true' and Exists('$(DotNetHostBinDir)')">
```
Comment (agocke) [inline eng/liveBuilds.targets]: @ViktorHofer This is the problematic part. By moving the host files to RuntimeFiles they ended up getting copied into the shared framework pack, when they were only supposed to be in the runtime pack. Keeping this in the Microsoft.NetCore.App package fixed the problem.

Comment (agocke) [reply eng/liveBuilds.targets]: Also, the changes in `externals.proj` basically replicate the behavior that was intended with the above change.

**Review by ViktorHofer — APPROVED** [review]
If it works, awesome 👍

@vitek-karas do you know if the host packages (Microsoft.NETCore.DotNetHost and Microsoft.NETCore.DotNetHostPolicy) are still consumed by anyone? Asking as this PR removes our own dependency on them.

**Comment by agocke** [comment]
@elinor-fung Mentioned that the host packages aren't directly depended on at the moment, but they're our vehicle for symbols for the hosting components. I'd want to bring the change through to stop building them separately, and confirm we still have symbols.

**Pull request merged**